A40-csds-support.md

```diff
@@ -0,0 +1,328 @@
+xDS Configuration Dump via Client Status Discovery Service in gRPC
+----
+* Author(s): lidizheng
+* Approver: markdroth
+* Status: In-Review
+* Implemented in: All languages
+* Last updated: 2021-02-19
+* Discussion at: https://groups.google.com/g/grpc-io/c/zL45YyxtJ08
+
+## Abstract
+
+[Client Status Discovery
+Service](https://github.com/envoyproxy/envoy/blob/main/api/envoy/service/status/v3/csds.proto)
+(CSDS) is a service that exposes xDS config of a given client. It’s commonly
+used to query control planes for the synced xDS config of a particular sidecar
+proxy. However, it can also be used to query an xDS-compliant application for
+its received xDS configuration. This doc proposes a solution to implement a CSDS
+service in gRPC, so our users can debug their service mesh easily.
+
+
+## Background
+
+Envoy [started](https://github.com/envoyproxy/envoy/pull/9383) the CSDS
+development process in Dec 2019. The CSDS API has been available since xDS v2,
+and it’s under active development. But, Envoy proxies do not support or
+understand this service, only the control plane does. Envoy already provides
+config dump via its [admin
+interface](https://www.envoyproxy.io/docs/envoy/latest/operations/admin) for
+years. There is no visible plan for Envoy to support serving CSDS directly as a
```

Comment:
Might be worth linking to https://github.com/envoyproxy/envoy/issues/13181.

Reply:
Good catch! I forgot the existence of this issue.